### ISSUE TITLE
Bluetooth-related kernel null pointer dereference

### ISSUE THREAD

**Issue by jrobsonchase** [NixOS/nixpkgs#300547]
### Describe the bug

Saw a stack trace on boot after updating my system nixpkgs to d8fe5e6c92d0d190646fb9f1056741a229980089. USB devices also seemed to stop working, even though they showed up in `lsusb`.

Upstream kernel bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=218651
Related Arch thread: https://bbs.archlinux.org/viewtopic.php?id=294292

Seems like this impacts 6.6.23 and 6.8.2.

I also noticed a hang on reboot which required me to hard shutdown. It seemed to get through killing all processes and unmounting filesystems, but then hung at the actual "shutdown".

Reverting to 2726f127c15a4cc9810843b96cad73c7eb39e443 (and thus kernel 6.6.22) resolved the issue.

### Screenshots

[dmesg.txt](https://github.com/NixOS/nixpkgs/files/14817586/dmesg.txt)

### Additional context

```
$ uname -a
Linux tarvos 6.6.23 #1-NixOS SMP PREEMPT_DYNAMIC Tue Mar 26 22:22:53 UTC 2024 x86_64 GNU/Linux
```

### Notify maintainers

@K900 
@Ma27 

### Metadata

Pretty sure this isn't terribly helpful:
```console
$ nix run nixpkgs#nix-info
system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.18.2, nixpkgs: /nix/store/cb1gs888vfqxawvc65q1dk6jzbayh3wz-source
```

My system nixpkgs is `github:NixOS/nixpkgs/d8fe5e6c92d0d190646fb9f1056741a229980089`.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jrobsonchase** [comment]
Can also confirm that [this patch](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=1c3366abdbe884) resolves the issue with kernel 6.6.23

**Comment by K900** [comment]
Patch is picked into 6.8.3: https://lore.kernel.org/stable/5400491d-e467-8a12-d7fb-6581efe86e1b@w6rz.net/T/#m168c890079426e833288a9f022868683a414d20c

So give it a few days.
